### ISSUE TITLE
Conflicting IAM Policy

### ISSUE THREAD

**Issue by lifelofranco** [terraform-aws-modules/terraform-aws-lambda#61]
Hi

I am deploying my lambda fxns into multiple regions. I ran into an issue on having duplicate IAM policies, since I have already deployed my lambda fxn to another region. Checked and the current code does not allow to manually update the policy name. It instead defaults to `"${function_name}-logs"` or `"${var.function_name}-dl"` I wouldn't want to change my function name to get around this.

```
resource "aws_iam_policy" "logs" {
  count = local.create_role && var.attach_cloudwatch_logs_policy ? 1 : 0

  name   = "${var.function_name}-logs"
  policy = data.aws_iam_policy_document.logs[0].json
}
```

A simple fix would be to change it to how you declared the IAM roles `name = coalesce(var.role_name, var.function_name)`. 
What do you think?

**Comment by simonweil** [comment]
I would really like this same feature but for another use case:

In the company I work for we have naming conventions for all resources created including policies.
It would be really great to have the possibility to allow passing names or a even an optional prefix and suffix for the Policy name instead of using the lambda name for it.

**Comment by antonbabenko** [comment]
Thanks @lifelofranco for opening this issue!

Fixed in #74, [v1.29.0](https://github.com/terraform-aws-modules/terraform-aws-lambda/releases/v1.29.0) has been just released.

I think it is enough to use the IAM role name as a prefix to the associated policy names just to avoid even more variables that can be specified in this module.

**Comment by lifelofranco** [comment]
Agree and thanks for this! 

Closing the issue.
